server/api.js

```diff
@@ -1,9 +1,59 @@
 import { Router } from "express";
+import pool from "./db";
 
 const router = Router();
+const bcrypt = require("bcrypt");
 
-router.get("/", (_, res) => {
-	res.json({ message: "Hello, world!" });
+router.get("/users", (req, res) => {
+	res.json("hello world");
 });
 
+router.post("/signup", async (req, res) => {
+	try {
+		//this hides the password
+		const hashedPassword = await bcrypt.hash(req.body.password, 10);
+		const newUser = {
+			firstName: req.body.firstName,
+			lastName: req.body.lastName,
+			username: req.body.username,
+			cohortId: req.body.cohortId,
+			password: hashedPassword,
+		};
+		//checking if username already exist
+		pool
+		.query("SELECT user_name FROM users WHERE user_name=$1", [newUser.username])
```

Comment:
This line is perfectly fine, but there is this: ``SELECT EXISTS(SELECT user_name FROM users WHERE user_name = $1);``

The syntax could be wrong, so you may want to research it. But this returns a boolean value based on the expression in the brackets. If it exists you'll get a TRUE returned else FALSE. But this line is perfectly fine it does the same thing haha